contrib/golang/filters/http/source/config.cc

```diff
@@ -0,0 +1,51 @@
+#include "contrib/golang/filters/http/source/config.h"
+
+#include "envoy/registry/registry.h"
+
+#include "contrib/golang/filters/http/source/common/dso/dso.h"
+#include "contrib/golang/filters/http/source/golang_filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Golang {
+
+Http::FilterFactoryCb GolangFilterConfig::createFilterFactoryFromProtoTyped(
+    const envoy::extensions::filters::http::golang::v3alpha::Config& proto_config,
+    const std::string&, Server::Configuration::FactoryContext& context) {
+
+  FilterConfigSharedPtr config = std::make_shared<FilterConfig>(proto_config);
+
+  handler_ = context.lifecycleNotifier().registerCallback(
+      Server::ServerLifecycleNotifier::Stage::PostInit,
+      [lib_id = config->soId(), lib_path = config->soPath()] {
+        Envoy::Dso::DsoInstanceManager::pub(lib_id, lib_path);
+        // ENVOY_LOG(info, "open golang library at postInit: {} {}", lib_id, lib_path);
+      });
+
+  return [&context, config](Http::FilterChainFactoryCallbacks& callbacks) {
+    auto filter =
+        std::make_shared<Filter>(context.grpcContext(), config, Filter::global_stream_id_++,
+                                 Dso::DsoInstanceManager::getDsoInstanceByID(config->soId()));
```

Comment:
How is error handling done if the DSO doesn't load? That part is not clear to me.